### DISCUSSION TITLE
Dark Reader Extension Icon should show if the site dark mode is enabled or disabled

### DISCUSSION THREAD

**Discussion by Prutal** [darkreader/darkreader#7772] [Ideas]
I need to know if the dark mode is enabled on the site I am currently browsing without clicking the icon
it would be nice if the icon can reflect that   🌞 \ 🌚 hopping to see this feature.

**Comment by Gusted** [comment]
> I need to know if the dark mode is enabled on the site I am currently browsing without clicking the icon

Well, I hopefully can expect from users that when they see a website, that they can see it's in dark mode or not.

**Reply by PianoMastR64** [reply]
If a website has its own custom dark mode turned on but Dark Reader is imposing _its_ dark mode instead, then this may not be obvious to the user without the suggested icon indicator.

**Comment by msucorey** [comment]
Hundred times yes.  I'll write the user story for this...

As a Dark Reader user, I want the Dark Reader Extension Icon to reflect whether I've toggled the current site (⌥⇧A), not the extension itself (⌥⇧D),

So that I can tell whether this site looks fine by itself or if I'm still having to use Dark Reader on this site.

Acceptance criteria:
On a given site, the Dark Reader Extension Icon changes from Darth Vader to Anakin as the user toggles the site from the keyboard ⌥⇧A or the extension dialog.

EDIT: Anakin h/t @victor871129

**Reply by victor871129** [reply]
The person with cool glasses is Anakin but young and not "dark" yet

**Comment by KBelmin** [comment]
I cannot believe we still do not have this option. So many website have their own dark mode and Dark Reader just does not recognize it -- Either fix that or please give us the option to show if Dark Reader has dark mode enabled - because sometimes the website has it enabled and it does not look right... Common..

**Reply by Myshor** [reply]
If you have some sites that has dark mode you can make some issue that Dark Mode is not detected and we can try to fix it.

We have already dark theme detection included in Dark Reader and Dark Reader shows if it's detected on site or not.
<img width="250" height="50" alt="obraz" src="https://github.com/user-attachments/assets/e64673dc-d322-43fe-9a23-788f1af24752" />

P.S. If there would be some badges in github for digging out some topic which was burried down in 2023 (last reply in this discussion) I think you would get one. 😋

**Reply by KBelmin** [reply]
> If you have some sites that has dark mode you can make some issue that Dark Mode is not detected and we can try to fix it.
> 
> We have already dark theme detection included in Dark Reader and Dark Reader shows if it's detected on site or not. <img alt="obraz" width="250" height="50" src="https://private-user-images.githubusercontent.com/56877029/611396610-e64673dc-d322-43fe-9a23-788f1af24752.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODIxNjE3ODAsIm5iZiI6MTc4MjE2MTQ4MCwicGF0aCI6Ii81Njg3NzAyOS82MTEzOTY2MTAtZTY0NjczZGMtZDMyMi00M2ZlLTlhMjMtNzg4ZjFhZjI0NzUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA2MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNjIyVDIwNTEyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1M2QxMzlhZDc1NjBjODg3NjljOTMwZDE2ZDNhODgwYzE5ODBjOWY4MjI4MDZmZWE0YWYyYWExMjIyNGNhOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.FuBHM-B8p1K-wJkJBEKqKrUSlg2jSqn9PngRstyqJLw">
> 
> P.S. If there would be some badges in github for digging out some topic which was burried down in 2023 (last reply in this discussion) I think you would get one. 😋

Believe me, if you give us the **badge** visible immediatelly on the top right bar from the extension

<img width="207" height="142" alt="image" src="https://github.com/user-attachments/assets/16c4bf02-95bd-439d-abac-91cc55828d18" />

instead of opening the menu (the point is to immediatelly know whether Dark Reader did not recognize dark mode from the website so we can turn it off for that site (for me its Ctrl + Spacebar) 
So having the badge change to maybe a red icon when its turned off for that specific page and green again when its turned on, would be golden -- I think more people should give their opinion on this.

I hope I was clear, ps. I love dark reader this is just constructive feedback -- I hope you give us this option, its much more used and needed than the color change when Dark Reader is completely off.

All the best and I stumbled upon this thread because I searched for a fix on google and found this haha :D

**Comment by victor871129** [comment]
It must be Darth Vader when turned on and no dark theme detected, Luke when turned off and Grogu when turned on and dark theme detectedOn 22 Jun 2026, at 3:57 PM, B ***@***.***> wrote:﻿

If you have some sites that has dark mode you can make some issue that Dark Mode is not detected and we can try to fix it.
We have already dark theme detection included in Dark Reader and Dark Reader shows if it's detected on site or not. 
P.S. If there would be some badges in github for digging out some topic which was burried down in 2023 (last reply in this discussion) I think you would get one. 😋

Believe me, if you give us the badge visible immediatelly on the top right bar from the extension
image.png (view on web)
instead of opening the menu (the point is to immediatelly know whether Dark Reader did not recognize dark mode from the website so we can turn it off for that site (for me its Ctrl + Spacebar)
So having the badge change to maybe a red icon when its turned off for that specific page and green again when its turned on, would be golden -- I think more people should give their opinion on this.
I hope I was clear, ps. I love dark reader this is just constructive feedback -- I hope you give us this option, its much more used and needed than the color change when Dark Reader is completely off.
All the best and I stumbled upon this thread because I searched for a fix on google and found this haha :D

**Reply by VodaJP** [reply]
Sorry it will be more confusing if it is Luke and not Anakin. Icon must be four states: child Anakin (current non-dark icon), adult Anakin with long hair, Darth Vader (current dark icon) and Darth Vader without mask

**Comment by Myshor** [comment]
@alexanderby Come take a look on last comments. Do you think we can take that requests in queue for future? 😉

**Comment by AntonioRedondo** [comment]
In 2020 a pull request tackling this issue was open (https://github.com/darkreader/darkreader/pull/3218, with related issue https://github.com/darkreader/darkreader/issues/2027). But ultimately it was closed without merging.

If I got a cent every time I have to open the Dark Reader icon pop-up to know whether the add-on is enabled or disabled for a given website I'd be millionaire by now! 😄

**Reply by KBelmin** [reply]
Oh I agree so much.
